src/sgame/sg_bot.cpp

```diff
@@ -586,28 +586,69 @@ void G_BotFill(bool immediately)
 	if (!immediately && level.time < nextCheck) {
 		return;  // don't check every frame to prevent warning spam
 	}
+
 	nextCheck = level.time + 2000;
+	//FIXME this function can actually be called before bots had time to connect
+	//  resulting in filling too many bots.
+	if ( level.matchTime < 2000 )
```

Comment:
I wonder if you could get rid of this if you changed the test to `level.time <= nextCheck` to avoid running at time 0 (and set immediately=false on the cvar callback).

Reply:
It seems what *can* be done here is to change the starting value of nextCheck to 2000, and remove this test, actually.
Simply removing the test leads to a segfault, which means it makes it easy to get a segfault if removing this test at all.

I think it would be an error to do that. The "solution" there might be ugly, but at least I think it's rather safe, and it actually works.